src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -547,130 +543,126 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
+      };
+      typeInfo.storedSettings = storedSettings;
+
+      this.trigger("trackUpdate", {
+        period: toExposedPeriod(period),
         trackType,
-        period: { id: period.id, start: period.start, end: period.end },
+        reason: "no-playable-representation",
       });
-      // The previous event trigger could have had side-effects, so we
-      // re-check if we're still mostly in the same state
-      if (this._isDisposed) {
-        return []; // Someone disposed the `TracksStore` on the previous side-effect
-      }
-    } else if (firstPlayableAdaptation === undefined) {
-      const noRepErr = new MediaError(
-        "NO_PLAYABLE_REPRESENTATION",
-        `No ${trackType} Representation can be played`,
-        { tracks: undefined },
-      );
-      noPlayableTrackToTrigger.push({
+      typeInfo.dispatcher?.updateTrack(storedSettings);
+    } else if (fallbackTrack === null && !noSourceMedia) {
+      this.trigger("noPlayableTrack", {
         trackType,
         period: { id: period.id, start: period.start, end: period.end },
       });
       // The previous event trigger could have had side-effects, so we
       // re-check if we're still mostly in the same state
       if (this._isDisposed) {
-        return []; // Someone disposed the `TracksStore` on the previous side-effect
+        return; // Someone disposed the `TracksStore` on the previous side-effect
       }
-      this.trigger("error", noRepErr);
-      this.dispose();
-      return [];
-    }
-    let typeInfo = getPeriodItem(this._storedPeriodInfo, period.id)?.[trackType];
-    if (isNullOrUndefined(typeInfo)) {
-      return noPlayableTrackToTrigger;
-    }
-    const selectedAdaptation = getPeriodItem(this._storedPeriodInfo, period.id)?.[
-      trackType
-    ]?.storedSettings;
-    if (selectedAdaptation === null) {
-      // The track type has been explicitly disabled, there is no need to select
-      // a new track as a fallback.
-      return noPlayableTrackToTrigger;
+      const fallbackBehavior = this.onTracksNotPlayableForType[trackType];
+      if (hasStoredSettingsChanged()) {
+        // The previous "noPlayableTrack" event trigger could have had side-effects, so we
+        // re-check if we're still mostly in the same state
+      } else if (fallbackBehavior === "continue") {
+        log.warn(`TS: No playable ${trackType}, continuing without ${trackType}`);
+        typeInfo.storedSettings = null;
+        this.trigger("trackUpdate", {
+          period: toExposedPeriod(period),
+          trackType,
+          reason: "no-playable-representation",
+        });
+      } else if (fallbackBehavior === "error") {
+        const noRepErr = new MediaError(
+          "NO_PLAYABLE_REPRESENTATION",
+          `No ${trackType} Representation can be played`,
+          { tracks: undefined },
+        );
+        this.trigger("error", noRepErr);
+      }
+    } else if (fallbackTrack === null && noSourceMedia) {
```

Comment:
I'm not sure I get that case how can we be in onNoPlayableRep for a type if there's no rep for that type?

Is it just for exhaustiveness and because "weird shit happens"? If so, maybe a log warn could put more attention into this weird case.

Or maybe I'm just missing something

Reply:
Yes I think you are right, but actually the function `onNoPlayableRepresentation` is also called when selecting an initial track. So when selecting an initial track you can have an audio / video only, and that condition is useful in this case. I could rename `onNoPlayableRepresentation` to `fallbackTrackForType` or something like that? what do you think?